jobscheduler/scheduler-component-shared/src/java/org/sakaiproject/component/app/scheduler/jobs/LBCSSFolderHideJob.java

```diff
@@ -0,0 +1,95 @@
+package org.sakaiproject.component.app.scheduler.jobs;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.quartz.DisallowConcurrentExecution;
+import org.quartz.Job;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+import org.sakaiproject.content.api.ContentCollectionEdit;
+import org.sakaiproject.content.api.ContentHostingService;
+import org.sakaiproject.db.api.SqlService;
+import org.sakaiproject.entity.api.ResourceProperties;
+import org.sakaiproject.entity.api.ResourcePropertiesEdit;
+import org.sakaiproject.event.api.UsageSessionService;
+import org.sakaiproject.exception.*;
+import org.sakaiproject.tool.api.Session;
+import org.sakaiproject.tool.api.SessionManager;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.util.List;
+
+/**
+ * Job to make LB-CSS folder hidden but contents accessible.
+ * LB-CSS folder should not be visible to access user, this job will hide all existing non-hidden LB-CSS folder.
+ * @author neelam
+ *
+ */
+@DisallowConcurrentExecution
+public class LBCSSFolderHideJob implements Job {
+
+	private SqlService sqlService;
+	private SessionManager sessionManager;
+
+	private static final Log LOG = LogFactory.getLog(LBCSSFolderHideJob.class);
+	
+	private ContentHostingService contentHostingService;
+
+	public void setSqlService(SqlService sqlService) {
+		this.sqlService = sqlService;
+	}
+
+	public void setSessionManager(SessionManager sessionManager) {
+		this.sessionManager = sessionManager;
+	}
+
+	public void setContentHostingService(ContentHostingService contentHostingService) {
+		this.contentHostingService = contentHostingService;
+	}
+
+	public void execute(JobExecutionContext context) throws JobExecutionException {
+		ContentCollectionEdit contentCollectionEdit = null;
+		ResourcePropertiesEdit resourceProperties = null;
+		Session session = null;
+		//sql to get collection ids for all LB-CSS folder
+		String sql = "SELECT COLLECTION_ID FROM CONTENT_COLLECTION WHERE COLLECTION_ID LIKE \"%LB-CSS/\" ";
+		LOG.debug("SQL to get collection Id for LB-CSS folder " + sql);
+		int totalFolders = 0 , foldersUpdated = 0, foldersAlreadySet = 0;
+		//set current user in session
+		try {
+			session = sessionManager.getCurrentSession();
+			session.setUserEid("admin");
+			session.setUserId("admin");
+			List<String> collection_ids = sqlService.dbRead(sql);
+			totalFolders = collection_ids.size();
+			LOG.info("Number of LB-CSS folders found is " + totalFolders);
+			for (String id : collection_ids) {
+				try {
+					contentCollectionEdit = contentHostingService.editCollection(id);
+					resourceProperties = contentCollectionEdit.getPropertiesEdit();
+				} catch (SakaiException exception) {
```

Comment:
I think you want this catch block at the end of the loop as if there is a problem getting the collection you don't want to try to do anything else with that resource.